archunit/src/main/java/com/tngtech/archunit/core/importer/JavaClassProcessor.java

```diff
@@ -324,6 +324,13 @@ public void visitMethodInsn(int opcode, String owner, String name, String desc,
             accessHandler.handleMethodInstruction(owner, name, desc);
         }
 
+        @Override
+        public void visitTypeInsn(int opcode, String type) {
+            if (opcode == Opcodes.INSTANCEOF) {
+                codeUnitBuilder.addInstanceOfCheck(JavaTypeImporter.createFromAsmObjectTypeName(type));
```

Comment:
As far as I can see, you can even pass in the `actualLineNumber` here, which will be really neat, because failure messages can report the exact location where the `instanceof` check has appeared! :smile: 
You might need something like `RawInstanceofCheck` though, to transport the info until you have all the classes imported...

Reply:
Seems to work, although I'm not completely sure why there is a difference between `lineNumber` and `sourceCodeLocation` in `Dependency`.